utils/githubUtils.js

```diff
@@ -5,16 +5,16 @@ import {
 } from "./customErrors.js";
 import { GITHUB_TOKEN } from "../config/constants.js";
 
+// Initialize Octokit client with the GitHub token
+const octokit = github.getOctokit(GITHUB_TOKEN);
+
 /**
  * Extracts relevant data from a GitHub Pull Request.
  * @returns {Promise<Object>} A promise that resolves to an object containing details of the pull request.
  * @throws {PullRequestDataExtractionError} Throws a custom error if data extraction fails.
  */
 export const extractPullRequestData = async () => {
   try {
-    // Initialize Octokit client with the GitHub token
-    const octokit = github.getOctokit(GITHUB_TOKEN);
-
```

Comment:
Ok with this! We should do the same for `createOrUpdateFile()`

Reply:
At last I going to keep this inside each function. Later I will include more modularization. That change was introduced I think in PR #16

Reply:
Sounds good. I'll mark this conversation as resolved.